src/librbd/image/CloneRequest.cc

```diff
@@ -213,9 +211,7 @@ void CloneRequest<I>::validate_parent() {
     close_parent();
     return;
   }
-  if (m_use_p_features) {
-    m_features = (p_features & ~RBD_FEATURES_IMPLICIT_ENABLE);
-  }
+  m_features = (p_features & ~RBD_FEATURES_IMPLICIT_ENABLE);
```

Comment:
I think this is what you really need:

```
if (m_use_p_features) {
  m_features = p_features;
}
m_features &= ~RBD_FEATURES_IMPLICIT_ENABLE;
```

Reply:
Right.

Reply:
I submit it in a standalone PR https://github.com/ceph/ceph/pull/37743.